Short circuits the image combining if diffuse and alpha textures are the same

tfili opened the pull request:
Some materials seem to have the diffuse and alpha textures set to the same file, so no need to combine them. Also added a test.

lilleyse commented on lib/loadMtl.js:
```diff
@@ -387,6 +387,10 @@ function createDiffuseAlphaTexture(diffuseTexture, alphaTexture, options) {
     }
 
+    if (diffuseTexture.path === alphaTexture.path) {
+        return diffuseTexture;
```
I think `diffuseTexture.transparent = true` needs to be set here in order for the blend mode to be set correctly. And add these lines to the new test to ensure that:

```
expect(material.alphaMode).toBe('BLEND');
expect(material.doubleSided).toBe(true);
```

tfili commented:
@lilleyse I added them. This should be good to go. It looks like in certain circumstances it wasn't using the returned texture to check transparency so I fixed that too.

lilleyse commented:
After testing and looking at the code closer I realized that https://github.com/AnalyticalGraphicsInc/obj2gltf/pull/205#discussion_r305466284 was actually an unnecesary suggestion and it was already setting the blend mode correctly because if sets transparent if an alpha texture exists. I think reverting back to https://github.com/AnalyticalGraphicsInc/obj2gltf/pull/205/commits/b0de7d1bc1373855b2b7197588d427b1e89f8383 is ok but adding the test change in https://github.com/AnalyticalGraphicsInc/obj2gltf/pull/205#discussion_r305466284.

Sorry for slowing this down.

tfili commented:
@lilleyse Changed it back but kept the tests. Everything still passes.

tfili commented:
@lilleyse Bump
